app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -110,48 +110,39 @@ fun MapScreen(
                           .withIconImage(resizedBitmap)
                     }
 
-                // This is a test code snippet that can uncommented for testing purposes as per the
-                // PR description
-
-                //                val parking1 = Point.fromLngLat(6.566, 46.519)
-                //                val parking2 = Point.fromLngLat(6.500,46.500)
-                //                val parking3 = Point.fromLngLat(6.600,46.530)
-                //
-                //                val listTest = listOf(parking1,parking2,parking3)
-                //
-                //                // Create PointAnnotationOptions for each parking with test tags
-                //                val pointAnnotationOptions = listTest.mapIndexed { index, point ->
-                //
-                //                    PointAnnotationOptions()
-                //                        .withPoint(point)
-                //                        .withIconImage(resizedBitmap)
-                //                }
-
                 // Add annotations to the annotation manager and display them on the map
                 pointAnnotationOptions.forEach { annotationManager.create(it) }
 
                 // Add a camera change listener to detect zoom changes
                 val cameraChangeListener = OnCameraChangeListener {
                   state.value = mapView.mapboxMap.cameraState.zoom
 
-                  PairScreenCoordinate = getScreenCoordinateFromPoint(mapView)
-
-                  parkingViewModel.getParkingsInRect(
-                      PairScreenCoordinate.second, PairScreenCoordinate.first)
-
-                  // Create PointAnnotationOptions for each parking
-                  pointAnnotationOptions =
-                      listOfParkings.value.map { parking ->
-                        PointAnnotationOptions()
-                            .withPoint(parking.location.center)
-                            .withIconImage(resizedBitmap)
-                      }
-
-                  // Clear all annotations from the annotation manager to avoid duplicates
-                  annotationManager.deleteAll()
-
-                  // Add annotations to the annotation manager and display them on the map
-                  pointAnnotationOptions.forEach { annotationManager.create(it) }
+                  // Get the top right and bottom left coordinates of the current view only when
+                  // what the user sees is outside the screen
+                  val (currentBottomLeft, currentTopRight) =
+                      getScreenCorners(mapView, useBuffer = false)
+                  if (!inBounds(
+                      currentBottomLeft, currentTopRight, loadedBottomLeft, loadedTopRight)) {
+                    Log.d("MapScreen", "Loading parkings in new view")
+                    // Get the buffered coordinates for loading parkings
+                    loadedBottomLeft = getScreenCorners(mapView, useBuffer = true).first
```

Comment:
It's an open question but wouldn't it be better to just compute once the function `getScreenCorners()` instead of 2 times here ?

Reply:
you are entirely right, fixed